src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -323,6 +328,78 @@ class spell_pri_divine_hymn : public SpellScript
     }
 };
 
+// 110744 - Divine Star
+struct areatrigger_pri_divine_star : AreaTriggerAI
+{
+    areatrigger_pri_divine_star(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger) {}
+
+    void OnUpdate(uint32 diff) override
+    {
+        _scheduler.Update(diff);
+    }
+
+    void OnUnitEnter(Unit* unit) override
+    {
+        if (Unit* caster = at->GetCaster())
+        {
+            if (std::find(_affectedUnits.begin(), _affectedUnits.end(), unit->GetGUID()) == _affectedUnits.end())
+            {
+                if (caster->IsValidAttackTarget(unit))
+                    caster->CastSpell(unit, SPELL_PRIEST_DIVINE_STAR_DAMAGE, CastSpellExtraArgs(TriggerCastFlags(TRIGGERED_IGNORE_GCD | TRIGGERED_IGNORE_CAST_IN_PROGRESS)));
+                else if (caster->IsValidAssistTarget(unit))
+                    caster->CastSpell(unit, SPELL_PRIEST_DIVINE_STAR_HEAL, CastSpellExtraArgs(TriggerCastFlags(TRIGGERED_IGNORE_GCD | TRIGGERED_IGNORE_CAST_IN_PROGRESS)));
+
+                _affectedUnits.push_back(unit->GetGUID());
+            }
+        }
+    }
+
+    void OnDestinationReached() override
+    {
+        if (Unit* caster = at->GetCaster())
+        {
+            _affectedUnits.clear();
+
+            if (!_returningToCaster)
+            {
+                _scheduler.Schedule(Milliseconds(), [this, caster](TaskContext task)
+                {
+                    if (!_returningToCaster || caster->GetDistance(_casterCurrentPosition) > 1.0f)
+                    {
+                        _returningToCaster = true;
+                        _casterCurrentPosition = caster->GetPosition();
+
+                        Movement::PointsArray returnSplinePoints;
+
+                        for (uint8 i = 0; i < 4; i++)
+                        {
+                            G3D::Vector3 returnPoint;
+                            returnPoint.x = (i < 2) ? at->GetPositionX() : caster->GetPositionX();
+                            returnPoint.y = (i < 2) ? at->GetPositionY() : caster->GetPositionY();
+                            returnPoint.z = (i < 2) ? at->GetPositionZ() : caster->GetPositionZ();
+
+                            returnSplinePoints.push_back(returnPoint);
+                        }
+
+                        at->InitSplines(returnSplinePoints, (at->GetDistance(caster) / 24) * 1000);
+                    }
+
+                    // NOTE: this can be adjusted, 100-150 was the closest the AT seems to update its spline to the caster's current position.
+                    task.Repeat(Milliseconds(100));
+                });
+            }
+            else
+                at->Remove();
```

Comment:
you can actually put that to the start of the method and check

if (_returningToCaster)
{
   at->Remove();
   return;
}

then you prevent clearing unneccessarily the _affectedUnits and stuff

Reply:
moved it somewhere else.